src/sdk/trace/span_processor.rs

```diff
@@ -269,6 +283,57 @@ impl BatchSpanProcessor {
             config: Default::default(),
         }
     }
+
+    /// Create a new batch processor builder and set the config value based on environment variables.
+    ///
+    /// If the value in environment variables is illegal, will fall back to use default value.
+    ///
+    /// Note that export batch size should be less than or equals to max queue size.
+    /// If export batch size is larger than max queue size, we will lower to be the same as max
+    /// queue size
+    pub fn from_env<E, S, SO, I, IO>(
+        exporter: E,
+        spawn: S,
+        interval: I,
+    ) -> BatchSpanProcessorBuilder<E, S, I>
+        where
+            E: exporter::trace::SpanExporter,
+            S: Fn(BatchSpanProcessorWorker) -> SO,
+            I: Fn(time::Duration) -> IO,
+    {
+        let mut config = BatchConfig::default();
+        let schedule_delay = std::env::var(OTEL_BSP_SCHEDULE_DELAY_MILLIS)
+            .and_then(|delay| u64::from_str(&delay).or(Ok(OTEL_BSP_SCHEDULE_DELAY_MILLIS_DEFAULT)))
+            .unwrap();
```

Comment:
This way will panic if the var is not set I think? Probably best to go with a version that doesn't have a straight `.unwrap` in it, those tend to make people nervous when they look at the code.

Reply:
Yeah you are right. Should have tested it with no env vars set I guess. Fixed it.